From adaf1fe744df8d37acdb8b157abc5034b8ea1073 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 13 Apr 2016 16:37:38 -0400
Subject: Remove (almost) no-op setup functions

These became practical no-ops when I removed reference to the loader
symbol lists. gcc will still emit code (and bss) for them though. No
functional change, just doing it to prove that nothing special happens
at submodule load.

Reviewed-by: Julien Cristau <jcristau@debian.org>
Signed-off-by: Adam Jackson <ajax@redhat.com>

diff --git a/src/alp_driver.c b/src/alp_driver.c
index 01e05dc..39802de 100644
--- src/alp_driver.c
+++ src/alp_driver.c
@@ -159,8 +159,6 @@ static int gd7556_MaxClocks[] = {  80100,  80100,  80100,  80100,  80100 };
 #define ALP_MINOR_VERSION 0
 #define ALP_PATCHLEVEL 0
 
-static MODULESETUPPROTO(alpSetup);
-
 static XF86ModuleVersionInfo alpVersRec =
 {
 	"cirrus_alpine",
@@ -181,20 +179,10 @@ static XF86ModuleVersionInfo alpVersRec =
  */
 _X_EXPORT XF86ModuleData cirrus_alpineModuleData = {
     &alpVersRec,
-    alpSetup,
+    NULL,
     NULL
 };
 
-static pointer
-alpSetup(pointer module, pointer opts, int *errmaj, int *errmin)
-{
-	static Bool setupDone = FALSE;
-	if (!setupDone) {
-		setupDone = TRUE;
-	}
-	return (pointer)1;
-}
-
 #endif /* XFree86LOADER */
 
 _X_EXPORT const OptionInfoRec *
diff --git a/src/lg_driver.c b/src/lg_driver.c
index 589d14e..c3fe142 100644
--- src/lg_driver.c
+++ src/lg_driver.c
@@ -165,8 +165,6 @@ static int LgLinePitches[4][11] = {
 #define LG_MINOR_VERSION 0
 #define LG_PATCHLEVEL 0
 
-static MODULESETUPPROTO(lgSetup);
-
 static XF86ModuleVersionInfo lgVersRec =
 {
 	"cirrus_laguna",
@@ -187,21 +185,10 @@ static XF86ModuleVersionInfo lgVersRec =
  */
 _X_EXPORT XF86ModuleData cirrus_lagunaModuleData = {
     &lgVersRec,
-    lgSetup,
+    NULL,
     NULL
 };
 
-static pointer
-lgSetup(pointer module, pointer opts, int *errmaj, int *errmin)
-{
-    static Bool setupDone = FALSE;
-    
-    if (!setupDone) {
-	setupDone = TRUE;
-    }
-    return (pointer)1;
-}
-
 #endif /* XFree86LOADER */
 
 _X_EXPORT const OptionInfoRec *
-- 
cgit v0.10.2

